platform/mod.rs

```diff
@@ -7,71 +7,20 @@
-pub use platform::linux::UnixSharedMemory as OsIpcSharedMemory;
-#[cfg(target_os="linux")]
-pub use platform::linux::UnixChannel as OsIpcChannel;
-#[cfg(target_os="linux")]
-pub use platform::linux::UnixSelectionResult as OsIpcSelectionResult;
-#[cfg(target_os="linux")]
-pub use platform::linux::OpaqueUnixChannel as OsOpaqueIpcChannel;
-#[cfg(target_os="linux")]
-pub use platform::linux::UnixOneShotServer as OsIpcOneShotServer;
+mod os {
+    #[cfg(all(not(feature = "inprocess"), target_os = "linux"))]
+    include!("linux/mod.rs");
 
-#[cfg(target_os="macos")]
-pub use platform::macos::channel;
-#[cfg(target_os="macos")]
-pub use platform::macos::MachReceiver as OsIpcReceiver;
-#[cfg(target_os="macos")]
-pub use platform::macos::MachSender as OsIpcSender;
-#[cfg(target_os="macos")]
-pub use platform::macos::MachReceiverSet as OsIpcReceiverSet;
-#[cfg(target_os="macos")]
-pub use platform::macos::MachSharedMemory as OsIpcSharedMemory;
-#[cfg(target_os="macos")]
-pub use platform::macos::MachChannel as OsIpcChannel;
-#[cfg(target_os="macos")]
-pub use platform::macos::MachSelectionResult as OsIpcSelectionResult;
-#[cfg(target_os="macos")]
-pub use platform::macos::OpaqueMachChannel as OsOpaqueIpcChannel;
-#[cfg(target_os="macos")]
-pub use platform::macos::MachOneShotServer as OsIpcOneShotServer;
+    #[cfg(all(not(feature = "inprocess"), target_os = "macos"))]
+    include!("macos/mod.rs");
 
-// Windows and Android use in-process mpsc channels IPC for now
-#[cfg(any(target_os="windows", target_os="android"))]
-pub use platform::inprocess::channel;
-#[cfg(any(target_os="windows", target_os="android"))]
-pub use platform::inprocess::MpscReceiver as OsIpcReceiver;
-#[cfg(any(target_os="windows", target_os="android"))]
-pub use platform::inprocess::MpscSender as OsIpcSender;
-#[cfg(any(target_os="windows", target_os="android"))]
-pub use platform::inprocess::MpscReceiverSet as OsIpcReceiverSet;
-#[cfg(any(target_os="windows", target_os="android"))]
-pub use platform::inprocess::MpscSharedMemory as OsIpcSharedMemory;
-#[cfg(any(target_os="windows", target_os="android"))]
-pub use platform::inprocess::MpscChannel as OsIpcChannel;
-#[cfg(any(target_os="windows", target_os="android"))]
-pub use platform::inprocess::MpscSelectionResult as OsIpcSelectionResult;
-#[cfg(any(target_os="windows", target_os="android"))]
-pub use platform::inprocess::OpaqueMpscChannel as OsOpaqueIpcChannel;
-#[cfg(any(target_os="windows", target_os="android"))]
-pub use platform::inprocess::MpscOneShotServer as OsIpcOneShotServer;
+    #[cfg(feature = "inprocess")]
+    include!("inprocess/mod.rs");
+}
 
-#[cfg(target_os="linux")]
-mod linux;
-#[cfg(target_os="macos")]
-mod macos;
-#[cfg(any(target_os="windows", target_os="android"))]
-mod inprocess;
+pub use self::os::{OsIpcChannel, OsIpcOneShotServer, OsIpcReceiver, OsIpcReceiverSet};
+pub use self::os::{OsIpcSelectionResult, OsIpcSender, OsIpcSharedMemory};
+pub use self::os::{OsOpaqueIpcChannel, channel};
```

Comment:
I'm not sure what's wrong with multiline use statements... But I can live with that I guess.

Still, I think it would be vastly preferable to group the types in some logical manner. As it is, it is (subjectively) confusing, and (objectively) a PITA if another entry ever needs to be added...

I guess that's bikeshedding though -- feel free to disregard :-)